### PR TITLE
ScopeExcludeReason: Align reasons with SPDX 2.2

### PR DESCRIPTION
**Pull request by tsteenbe**
SPDX 2.2 offers new relationships that better match the reasons available for scopes excludes.

See https://github.com/spdx/spdx-spec/blob/development/v2.2/chapters/7-relationships-between-SPDX-elements.md

Signed-off-by: Thomas Steenbergen <thomas.steenbergen@here.com>

### CODE REVIEW

File: .ort.yml [hunk 1]
```diff
@@ -74,10 +74,10 @@ excludes:
     comment: "Licenses contained in this directory are used for testing and do not apply to the OSS Review Toolkit."
   scopes:
   - name   : "devDependencies"
-    reason : "BUILD_TOOL_OF"
+    reason : "DEV_DEPENDENCY_OF"
```
Comment (sschuberth) [inline .ort.yml]: Commit message: s/ScopeExludeReason/ScopeExcludeReason/

File: model/src/main/kotlin/config/ScopeExcludeReason.kt [hunk 2]
```diff
@@ -24,19 +24,48 @@ package com.here.ort.model.config
  */
 enum class ScopeExcludeReason {
     /**
+     * @deprecated
```
Comment (sschuberth) [inline model/src/main/kotlin/config/ScopeExcludeReason.kt]: To mark something as deprecated in Kotlin, use `@Deprecated` (captitalized) as part of the code, not the docs, and name the new symbol to use. Like:

    @Deprecated(
        message = "Use SPDX-2.2-style enum value instead.",
        replaceWith = ReplaceWith(
            expression = "ScopeExcludeReason.BUILD_DEPENDENCY_OF",
            imports = ["com.here.ort.model.config.ScopeExcludeReason"]
        )
    )
    BUILD_TOOL_OF,

File: .ort.yml (hunk 1 above)
Comment (tsteenbe) [reply .ort.yml]: Fixed

File: model/src/main/kotlin/config/ScopeExcludeReason.kt (hunk 2 above)
Comment (tsteenbe) [reply model/src/main/kotlin/config/ScopeExcludeReason.kt]: Fixed

File: docs/examples/go-mod.ort.yml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 excludes:
   scopes:
     - pattern: "all"
-      reason: "BUILD_TOOL_OF"
+      reason: "DEV_DEPENDENCY_OF"
```
Comment (sschuberth) [inline docs/examples/go-mod.ort.yml]: Shouldn't this be `BUILD_DEPENDENCY_OF`?

File: model/src/main/kotlin/config/ScopeExcludeReason.kt [hunk 4]
```diff
@@ -27,16 +27,63 @@ enum class ScopeExcludeReason {
...
+    TEST_TOOL_OF,
+
+    /**
+     * The scope only contains packages used for testing which are not included in distributed build.
+     */
+    TEST_DEPENDENCY_OF,
+
+    /**
+     * The scope only contains packages that have to be provided by the user
+     * during the execution of the artifacts but are not included in distributed build artifacts.
```
Comment (sschuberth) [inline model/src/main/kotlin/config/ScopeExcludeReason.kt]: Nit: Line wrapping looks like the first line still has room for more words.

File: docs/examples/go-mod.ort.yml (hunk 3 above)
Comment (fviernau) [reply docs/examples/go-mod.ort.yml]: comment from @tsteenbe  : dev dependency of is correct as Go Mod to my knowledge mixes test and build deps so safest option is dev dependency of.

**Comment by sschuberth** [comment]
Regarding the build failure, as we don't allow compiler warning, we should ourselves use the non-deprecated enum values in our code.

File: model/src/main/kotlin/config/ScopeExcludeReason.kt (hunk 4 above)
Comment (fviernau) [reply model/src/main/kotlin/config/ScopeExcludeReason.kt]: Done.

**Pull request merged**